addons/project_timesheet_holidays/models/hr_holidays.py

```diff
@@ -32,6 +34,12 @@ def _compute_timesheet_generate(self):
         for leave_type in self:
             leave_type.timesheet_generate = leave_type.timesheet_task_id and leave_type.timesheet_project_id
 
+    @api.depends('timesheet_task_id', 'timesheet_task_id.project_id')
+    def _compute_timesheet_project_id(self):
+        for leave_type in self:
+            if not leave_type.timesheet_project_id:
+                leave_type.timesheet_project_id = leave_type.timesheet_task_id.project_id
```

Comment:
What do we do in the case in which we have a project and a task set on a leave type and the task is linked to another project than the one set on the leave type? 🤔

Reply:
Hello @xavierbol 
Skip this point from spec by @aju-odoo.
Thanks!